UnityProject/Assets/Scripts/Items/Weapons/SawnOff.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.Collections.Generic;
+using Items;
+using AddressableReferences;
+using Mirror;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.Serialization;
+using Weapons;
+
+[RequireComponent(typeof(ItemAttributesV2))]
+[RequireComponent(typeof(Gun))]
+public class SawnOff : MonoBehaviour, ICheckedInteractable<InventoryApply>
```

Comment:
a little bit of a nitpick, but for me a good component name has "able" at the end to inform adding the component will add a particular functionality to whatever prefab you add attach it to. Also this one is specific for weapons, maybe ``SawoffableWeapon``? If that makes any sense in English.